nomad/blocked_evals.go

```diff
@@ -734,28 +746,36 @@ func (b *BlockedEvals) prune(stopCh <-chan struct{}) {
 		select {
 		case <-stopCh:
 			return
-		case <-ticker.C:
-			b.pruneUnblockIndexes()
+		case t := <-ticker.C:
+			cutoff := t.UTC().Add(-1 * pruneThreshold)
+			b.pruneUnblockIndexes(cutoff)
+			b.stats.prune(cutoff)
 		}
 	}
 }
 
 // pruneUnblockIndexes is used to prune any tracked entry that is excessively
 // old. This protects againsts unbounded growth of the map.
-func (b *BlockedEvals) pruneUnblockIndexes() {
+func (b *BlockedEvals) pruneUnblockIndexes(cutoff time.Time) {
 	b.l.Lock()
 	defer b.l.Unlock()
 
 	if b.timetable == nil {
 		return
 	}
 
-	cutoff := time.Now().UTC().Add(-1 * pruneThreshold)
 	oldThreshold := b.timetable.NearestIndex(cutoff)
-
 	for key, index := range b.unblockIndexes {
 		if index < oldThreshold {
 			delete(b.unblockIndexes, key)
 		}
 	}
 }
+
+// pruneStats is used to prune any zero value stats that are excessively old.
+func (b *BlockedEvals) pruneStats(cutoff time.Time) {
```

Comment:
This doesn't appear to have any non-test callers. Is it just to help facilitate test assertions? If so, let's note that in the docstring.

Reply:
It's used in the `prune` method a few lines up: https://github.com/hashicorp/nomad/pull/10454/files#diff-313ccb0614ac40b9897c65c87594666b2f6f92f95460b01fb27622c837e5a1ceR752

I used hooked this into the same mechanism that was being used to prune old index eval index tracking.

Reply:
Oh sorry. I just realized that I forgot to actually call this method....the line I linked was supposed to call it instead of `b.stats.prune()` directly since this method acquires a lock. Fixing it.